apps/cms/src/customize/graphql/mutation/inviteTest.ts

```diff
@@ -0,0 +1,210 @@
+    .query('api::test.test')
+    .findOne({ where: { id: testId } });
+
+  if (!testTemplate) {
+    return true;
+  }
+  return;
+};
+
+const getNonExistingCandidates = async (emails: string[]) => {
+  try {
+    const nonExistCandidates: Candidate[] = [];
+    await Promise.all(
+      emails.map(async (email: string) => {
+        const candidate = await strapi.db
+          .query('api::candidate.candidate')
+          .findOne({ where: { email } });
+
+        if (!candidate) {
+          nonExistCandidates.push({
+            firstName: email.split('@')[0],
+            email,
+            publishedAt: new Date().toISOString(),
+          });
+        }
+      })
+    );
+
+    return nonExistCandidates;
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const createManyCandidate = async (listCandidateToCreate: Candidate[]) => {
+  try {
+    if (listCandidateToCreate.length > 0) {
+      await strapi.db.query('api::candidate.candidate').createMany({
+        data: listCandidateToCreate,
+      });
+    }
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const createIndividualTests = async (emails: string[], testId: string) => {
+  try {
+    await Promise.all(
+      emails.map(async (email: string) => {
+        // If current candidate already invited to inviting test - do nothing
+        const isIndTestExist = await strapi.db
+          .query('api::individual-test.individual-test')
+          .findOne({
+            where: {
+              $and: [{ candidate: { email }, test: { id: testId } }],
+            },
+          });
+
+        if (!isIndTestExist) {
+          const currentCandidate = await strapi.db
+            .query('api::candidate.candidate')
+            .findOne({ where: { email } });
+
+          // todo currently createMany doesn't work correctly, find a way to use
+          await strapi.db.query('api::individual-test.individual-test').create({
+            data: {
+              test: testId,
+              candidate: currentCandidate.id,
+              publishedAt: new Date().toISOString(),
+            },
+          });
+        }
+      })
+    );
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const getIndTests = async (emails: string[], testId: string) => {
+  try {
+    const listOfIndTest = await Promise.all(
+      emails.map(async (email: string) => {
+        const indTest = await strapi.db
+          .query('api::individual-test.individual-test')
+          .findOne({
+            where: {
+              $and: [{ candidate: { email }, test: { id: testId } }],
+            },
+          });
+
+        const token = getService('jwt').issue({
+          individualTestId: indTest.id,
+          SECRET_KEY: process.env.JWT_SECRET,
+        });
+
+        return {
+          email,
+          // todo replace with real link-test
+          linkTestInvite: `http://localhost:4200/test/${testId}?token=${token}`,
```

Comment:
should have ENV for it `TEST_PAGE_URL=http://localhost:4200/test`  (we don't need the `testId`)
linkTestInvite: `${TEST_PAGE_URL}?token=${token}`